github_connector/models/github.py

```diff
@@ -110,33 +122,48 @@ def get_by_url(self, url, call_type, data=False):
         _logger.info("Calling %s" % url)
         for i in range(self.max_try):
             try:
+                if self.token:
+                    headers = {'Authorization': 'token {}'.format(self.token)}
+
+                    def get(u):
+                        return requests.get(u, headers=headers)
+
+                    def post(u, d):
+                        return requests.get(u, headers=headers, data=d)
+                else:
+                    auth = HTTPBasicAuth(self.login, self.password)
+
+                    def get(u):
+                        return requests.get(u, auth=auth)
+
+                    def post(u, d):
+                        return requests.get(u, auth=auth, data=d)
+
                 if call_type == 'get':
-                    response = requests.get(
-                        url, auth=HTTPBasicAuth(self.login, self.password))
+                    response = get(url)
                     break
                 elif call_type == 'post':
                     json_data = json.dumps(data)
-                    response = requests.post(
-                        url, auth=HTTPBasicAuth(self.login, self.password),
-                        data=json_data)
+                    response = post(url, json_data)
                     break
             except Exception as err:
-                _logger.warning(
+                _logger.exception(
```

Comment:
are you sure, it is "exception". Not better _logger.error ?

Reply:
warning is back ;)